src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -605,7 +606,12 @@ private User impersonate(final RestRequest request, final User originalUser) thr
             final boolean isDebugEnabled = log.isDebugEnabled();
             // loop over all http/rest auth domains
             for (final AuthDomain authDomain : restAuthDomains) {
+
                 final AuthenticationBackend authenticationBackend = authDomain.getBackend();
+                // Skip over the OnBehalfOfAuthenticator since it is not compatible for user impersonation
+                if (authDomain.getHttpAuthenticator() instanceof OnBehalfOfAuthenticator) {
```

Comment:
I don't follow this comment--I would clarify with Ryan or Craig what was meant here.

Reply:
Yes, DC is correct. The obo auth backend should not be used for impersonate a user.